server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocatorTests.java

```diff
@@ -539,10 +539,29 @@ private static IndexMetadata.Builder anIndex(String name, Settings.Builder setti
     private static ClusterState stateWithStartedIndices(IndexMetadata.Builder... indices) {
         var metadataBuilder = Metadata.builder();
         var routingTableBuilder = RoutingTable.builder(TestShardRoutingRoleStrategies.DEFAULT_ROLE_ONLY);
-        for (var index : indices) {
-            var build = index.build();
-            metadataBuilder.put(build, false);
-            routingTableBuilder.addAsNew(build);
+        if (randomBoolean()) {
+            // allocate all shards from scratch
+            for (var index : indices) {
```

Comment:
more naming suggestions

```suggestion
            for (var indexMetadataBuilder : indexMetadataBuilders) {
```